esphome/components/opentherm/switch/__init__.py

```diff
@@ -0,0 +1,72 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import switch
+from esphome.const import (
+    ICON_RADIATOR,
+)
+from ...opentherm import (
+    OpenThermComponent,
+    CONF_OPENTHERM_ID,
+)
+from .. import opentherm
+
+CustomSwitch = opentherm.class_("CustomSwitch", switch.Switch, cg.Component)
+
+CONF_CH_ENABLED = "ch_enabled"
+CONF_CH_2_ENABLED = "ch_2_enabled"
+CONF_DHW_ENABLED = "dhw_enabled"
+CONF_COOLING_ENABLED = "cooling_enabled"
+CONF_OTC_ACTIVE = "otc_active"
+
+ICON_WATER_BOILER = "mdi:water-boiler"
+ICON_SNOWFLAKE = "mdi:snowflake"
+ICON_THERMOMETER_AUTO = "mdi:thermometer-auto"
+
+TYPES = [
+    CONF_CH_ENABLED,
+    CONF_CH_2_ENABLED,
+    CONF_DHW_ENABLED,
+    CONF_COOLING_ENABLED,
+    CONF_OTC_ACTIVE,
+]
+
+CONFIG_SCHEMA = cv.All(
+    cv.Schema(
+        {
+            cv.GenerateID(CONF_OPENTHERM_ID): cv.use_id(OpenThermComponent),
+            cv.Optional(CONF_CH_ENABLED): switch.switch_schema(
+                class_=CustomSwitch,
```

Comment:
The key for `class_` is not required as its the first and only positional argument
```suggestion
                CustomSwitch,
```